paymentsheet/src/main/java/com/stripe/android/link/LinkControllerViewModel.kt

```diff
@@ -432,18 +449,14 @@ internal class LinkControllerViewModel @Inject constructor(
 
     private suspend fun createPaymentMethod(): Result<PaymentMethod> {
         val state = _state.value
-        val configuration = requireConfiguration(state)
+        val component = requireLinkComponent(state)
             .getOrElse { return Result.failure(it) }
+        val configuration = component.configuration
         val paymentMethod = state.selectedPaymentMethod
-        val account = _account.value
-
-        if (paymentMethod == null || account == null) {
-            return Result.failure(IllegalStateException("Invalid state"))
-        }
+            ?: return Result.failure(IllegalStateException("No selected payment method"))
 
         return if (configuration.passthroughModeEnabled) {
-            linkRepository.sharePaymentDetails(
-                consumerSessionClientSecret = account.clientSecret,
+            component.linkAccountManager.sharePaymentDetails(
```

Comment:
👀 Not strictly necessary but I'm using higher-level `LinkAccountManager` instead of lower-level `LinkRepository` for these calls.